cmd/httpclient/main.go

```diff
@@ -67,26 +68,41 @@ func mainfunc() (err error) {
 		fmt.Printf("\nExamples:\n")
 		fmt.Printf("%s\n", "  ./httpclient -dns-transport system ...")
 		fmt.Printf("%s\n", "  ./httpclient -dns-transport godns ...")
-		fmt.Printf("%s\n", "  ./httpclient -dns-transport doh ...")
-		fmt.Printf("%s\n", "  ./httpclient -dns-transport dot ...")
-		fmt.Printf("%s\n", "  ./httpclient -dns-transport tcp ...")
-		fmt.Printf("%s\n", "  ./httpclient -dns-transport udp [-dns-udp-server <addr>:<port>] ...")
+		fmt.Printf("%s\n", "  ./httpclient -dns-transport doh [-dns-server <url>] ...")
+		fmt.Printf("%s\n", "  ./httpclient -dns-transport dot [-dns-server <domain>] ...")
+		fmt.Printf("%s\n", "  ./httpclient -dns-transport tcp [-dns-server <addr>:<port>] ...")
+		fmt.Printf("%s\n", "  ./httpclient -dns-transport udp [-dns-server <addr>:<port>] ...")
 		fmt.Printf("\nWe'll select a suitable backend for each transport. Note\n")
 		fmt.Printf("that this only works on Unix.\n")
 		return nil
 	}
+
+	strDNSServer = *flagDNSServer
```

Comment:
you don't need this assignment; you can use and assign `*flagDNSServer` directly.

Reply:
It will look like this:
```go
	} else if *flagDNSTransport == "udp" {
		if *flagDNSServer == "" {
			err = client.ConfigureDNS("udp",  "1.1.1.1:53")
		} else {
			err = client.ConfigureDNS("udp", *flagDNSServer)
		}
	}
```
Looks good?

Reply:
How about:

```Go
		if *flagDNSServer == "" {
			*flagDNSServer = "8.8.8.8:53"
		}
```

?

Reply:
We can't change `FlagSet` stract (`flag` package) values unless we use the `flag.StringVar()` in the `init()` function and give it the `&strDNSServer` beforehand.

Reply:
Try it

Reply:
done

Reply:
Oh, sorry :-). I guess I was too brief. I meant that AFAIK using `*flagDNSServer` as I suggested works and I have been using it in `main_test.go`. Why do you think it won't work?

Reply:
:)) I see. But my first test was the way you said above. But it did cause failures in `main_test.go` which was unclear. So I had chosen another method.

Reply:
How was the test failing? I think we should try to make it work, because it seems better than introducing a new variable or unnecessary `ifs` thnn then we need to cover with tests.

Reply:
```
{"Connect":{"ConnID":64,"Duration":0,"Error":null,"LocalAddress":"64","Network":"godns-pseudo-conn","RemoteAddress":"64","Time":991331}}
{"Write":{"ConnID":64,"Duration":568,"Error":null,"NumBytes":37,"Time":1009351}}
{"DNSQuery":{"ConnID":64,"Message":{"Data":"aisBAAABAAAAAAAABG9vbmkCaW8HZXhhbXBsZQNjb20AAAEAAQ=="},"Time":1014544}}
{"Read":{"ConnID":64,"Duration":9460,"Error":{"Op":"parse","URL":"1.1.1.1:53","Err":{}},"NumBytes":0,"Time":1029046}}
{"Close":{"ConnID":64,"Duration":138,"Error":null,"Time":1051049}}
{"Resolve":{"Addresses":null,"ConnID":48,"Duration":949186,"Error":{"Err":"parse 1.1.1.1:53: first path segment in URL cannot contain colon","Name":"ooni.io","Server":"8.8.8.8:53","IsTimeout":false,"IsTemporary":false,"IsNotFound":false},"Hostname":"ooni.io","Time":1074474}}
2019/10/04 01:07:35 client.Get failed (error: Get https://ooni.io/: lookup ooni.io on 8.8.8.8:53: parse 1.1.1.1:53: first path segment in URL cannot contain colon)
2019/10/04 01:07:35 cannot fetch specific URL (error: Get https://ooni.io/: lookup ooni.io on 8.8.8.8:53: parse 1.1.1.1:53: first path segment in URL cannot contain colon)
--- FAIL: TestDoHTransport (0.00s)
    main_test.go:84: Get https://ooni.io/: lookup ooni.io on 8.8.8.8:53: parse 1.1.1.1:53: first path segment in URL cannot contain colon
2019/10/04 01:07:35 cannot configure DNS transport (error: invalid -dns-transport argument)
2019/10/04 01:07:35 client.Get failed (error: parse 	: net/url: invalid control character in URL)
2019/10/04 01:07:35 cannot fetch specific URL (error: parse 	: net/url: invalid control character in URL)
FAIL
coverage: 94.7% of statements
exit status 1
FAIL	github.com/ooni/netx/cmd/httpclient	8.427s
```
I don't know :(

Reply:
Can you push your change so that I can try it? It says `"parse 1.1.1.1:53: first path segment in URL cannot contain colon"`, so you're probably passing some invalid value to DoH?

Reply:
> Can you push your change so that I can try it?

Done
> so you're probably passing some invalid value to DoH?

No

Reply:
it says: `cannot configure DNS transport (error: invalid -dns-transport argument)`
so:
https://github.com/ooni/netx/blob/5bd750acf2f661559023b617f2ab54ca4a4f5fb7/cmd/httpclient/main.go#L106

So flagDNSTransport has no value. :)